src/mixxx.cpp

```diff
@@ -829,9 +792,38 @@ void MixxxMainWindow::finalize() {
     // at exit.
     m_pSettingsManager->save();
 
+    // Check for leaked ControlObjects and give warnings.
+    {
+        QList<QSharedPointer<ControlDoublePrivate>> leakedControls =
+                ControlDoublePrivate::takeAllInstances();
+        VERIFY_OR_DEBUG_ASSERT(leakedControls.isEmpty()) {
+            qWarning()
+                    << "The following"
+                    << leakedControls.size()
+                    << "controls were leaked:";
+            for (auto pCDP : leakedControls) {
+                ConfigKey key = pCDP->getKey();
+                qWarning() << key.group << key.item << pCDP->getCreatorCO();
+                // Deleting leaked objects helps to satisfy valgrind.
+                // These delete calls could cause crashes if a destructor for a control
+                // we thought was leaked is triggered after this one exits.
+                // So, only delete so if developer mode is on.
+                if (CmdlineArgs::Instance().getDeveloper()) {
+                    pCDP->deleteCreatorCO();
+                }
+            }
+        }
+        // Finally drop all shared pointers by exiting this scope
+    }
+
     Sandbox::shutdown();
 
     qDebug() << t.elapsed(false).debugMillisWithUnit() << "deleting SettingsManager";
+    {
+        auto* pSettingsManager = m_pSettingsManager.get();
+        m_pSettingsManager = nullptr; // reset parented_ptr
+        delete pSettingsManager;
+    }
```

Comment:
Why do we use a parented_ptr if we delete it manually anyway?

Reply:
If we don't use the Qt object hierarchy then we should consequently not parent this object at all. I only tried to preserve the original semantics, because I didn't investigate why it has been designed this way.

Reply:
Fixed.